tools/checky/src/main.rs

```diff
@@ -0,0 +1,51 @@
+use std::path::PathBuf;
+
+use checks::check;
+use clap::Parser;
+use color_eyre::{eyre::Context, Result};
+use rust_file::RustFile;
+use walkdir::WalkDir;
+
+mod checks;
+mod rust_file;
+mod syn_context;
+
+#[derive(Parser, Debug)]
+struct Arguments {
+    paths: Vec<PathBuf>,
+}
+
+fn main() -> Result<()> {
+    let arguments = Arguments::parse();
+
+    for path in arguments.paths {
+        for entry in WalkDir::new(&path) {
+            let entry =
+                entry.wrap_err_with(|| format!("failed to walk directory {}", path.display()))?;
+
+            if !entry.file_type().is_file() {
+                continue;
+            }
+
+            if !entry
+                .path()
+                .extension()
+                .map(|extension| extension == "rs")
+                .unwrap_or_default()
+            {
+                continue;
+            }
```

Comment:
If I provide a single non-rust file, I'd expect this tool to fail. Currently it silently runs to completion.

e.g.
```
cargo run --manifest-path tools/checky/Cargo.toml -- Cargo.toml
```

Reply:
Do you have a suggestion how to implement that without adding a lot of complexity in this file?